src/js/extend.js

```diff
@@ -27,9 +25,17 @@ import _inherits from '@babel/runtime/helpers/inherits';
  *           The new class with subClassMethods that inherited superClass.
  */
 const extend = function(superClass, subClassMethods = {}) {
-  let subClass = function() {
-    superClass.apply(this, arguments);
-  };
+  let subClass;
+
+  // If the provided super class is a native ES6 class,
+  // make the sub class one as well.
+  if (/^class/.test(superClass.toString())) {
```

Comment:
Since Video.js components are now un-transpiled ES6 classes, `extend()` needs to be able to handle them since `videojs.extend(videojs.getComponent('Component'), {...})` is the most common usecase for this method